File: backend/CodeyBE.Contracts/Entities/LessonGroup.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeyBE.Contracts.Entities
{
    public class LessonGroup
    {
        [BsonElement("_id")]
        public ObjectId Id { get; set; }
        [BsonElement("name")]
        public string Name { get; set; } = string.Empty;
        [BsonElement("tips")]
        [BsonIgnoreIfNull]
        public string? Tips { get; set; }
        [BsonElement("id")]
        public int PrivateId { get; set; }
        [BsonElement("order")]
        public int Order { get; set; }
        [BsonElement("lessons")]
        public List<int> LessonIds { get; set; } = [];
        [BsonElement("adaptive")]
        [BsonIgnoreIfNull]
        public bool? Adaptive { get; set; }
        [BsonElement("recommendedLevel")]
        [BsonIgnoreIfNull]
        public double? RecommendedLevel { get; set; }

    }
}

File: codey/lib/models/entities/lesson_group.dart
class LessonGroup {
  final int id;
  String name;
  String? tips;
  List<int> lessons;
  int order;
  bool adaptive;
  int courseId;

  LessonGroup({
    required this.id,
    required this.name,
    this.tips,
    required this.lessons,
    required this.order,
    this.adaptive = false,
    required this.courseId,
  });

  factory LessonGroup.fromJson(Map<String, dynamic> json) {
    return LessonGroup(
      id: json['privateId'],
      name: json['name'],
      tips: json['tips'],
      lessons: json['lessonIds'].cast<int>(),
      order: json['order'],
      adaptive: json['adaptive'] ?? false,
      courseId: json['courseId'],
    );
  }

  toJson() {
    return {
      'privateId': id,
      'name': name,
      'tips': tips,
      'lessonIds': lessons,
      'order': order,
      'adaptive': adaptive,
      'courseId': courseId,
    };
  }
}
